llms/hf_llm/convert.py

```diff
@@ -106,63 +170,11 @@ def upload_to_hub(path: str, name: str, hf_path: str):
 
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser(
-        description="Convert Hugging Face model to MLX format"
-    )
-    parser.add_argument(
-        "--hf-path",
-        type=str,
-        help="Path to the Hugging Face model.",
-    )
-    parser.add_argument(
-        "--mlx-path",
-        type=str,
-        default="mlx_model",
-        help="Path to save the MLX model.",
-    )
-    parser.add_argument(
-        "-q",
-        "--quantize",
-        help="Generate a quantized model.",
-        action="store_true",
-    )
-    parser.add_argument(
-        "--q-group-size",
-        help="Group size for quantization.",
-        type=int,
-        default=64,
-    )
-    parser.add_argument(
-        "--q-bits",
-        help="Bits per weight for quantization.",
-        type=int,
-        default=4,
-    )
-    parser.add_argument(
-        "--dtype",
-        help="Type to save the parameters, ignored if -q is given.",
-        type=str,
-        choices=["float16", "bfloat16", "float32"],
-        default="float16",
-    )
-    parser.add_argument(
-        "--upload-name",
-        help="The name of model to upload to Hugging Face MLX Community",
-        type=str,
-        default=None,
-    )
-    parser.add_argument(
-        "-l",
-        "--local",
-        action="store_true",
-        help="Whether the hf-path points to a local filesystem.",
-        default=False,
-    )
-
+    parser = configure_parser()
     args = parser.parse_args()
 
     print("[INFO] Loading")
-    weights, config, tokenizer = fetch_from_hub(args.hf_path, args.local)
```

Comment:
The local flag is not really needed. We can check if the path exists or not to determine if it's a local path.

Reply:
You're right, that's much nicer!